src/Encoder/Parser/Parser.php

```diff
@@ -132,6 +133,9 @@ private function parseData($data)
             if (is_array($data) === true) {
                 $isOriginallyArrayed = true;
                 $schema = $this->container->getSchema(reset($data));
+            } elseif ($data instanceof Traversable) {
+                $isOriginallyArrayed = true;
+                $schema = $this->container->getSchema($data[0]);
```

Comment:
It only works because an array is inside `$data` during the test.

I'll read the falcon docs and try to come up to some solution.